diff -rupN capstone-3.0.5-rc1/MCInst.c capstone/MCInst.c
--- capstone-3.0.5-rc1/MCInst.c	2016-07-18 08:28:04.000000000 +0300
+++ capstone/MCInst.c			2017-02-12 16:01:10.000000000 +0300
@@ -3,6 +3,7 @@
 
 #if defined(CAPSTONE_HAS_OSXKERNEL)
 #include <libkern/libkern.h>
+#include <kern/debug.h>
 #else
 #include <stdio.h>
 #include <stdlib.h>
@@ -152,6 +153,13 @@ void MCOperand_CreateReg0(MCInst *mcInst
 	MCOperand *op = &(mcInst->Operands[mcInst->size]);
 	mcInst->size++;
 
+#if defined(CAPSTONE_HAS_OSXKERNEL)
+	// Just in case because once we got some weird panic here
+	if (mcInst->size >= 48) {
+		panic("mcInst >= 48");
+	}
+#endif
+	
 	op->Kind = kRegister;
 	op->RegVal = Reg;
 }
diff -rupN capstone-3.0.5-rc1/cs.c capstone/cs.c
--- capstone-3.0.5-rc1/cs.c	2016-07-18 08:28:04.000000000 +0300
+++ capstone/cs.c			2017-02-12 15:55:58.000000000 +0300
@@ -322,7 +322,6 @@ static void fill_insn(struct cs_struct *
 #ifndef CAPSTONE_DIET
 	// fill in mnemonic & operands
 	// find first space or tab
-	sp = buffer;
 	mnem = insn->mnemonic;
 	for (sp = buffer; *sp; sp++) {
 		if (*sp == ' '|| *sp == '\t')
diff -rupN capstone-3.0.5-rc1/utils.h capstone/utils.h
--- capstone-3.0.5-rc1/utils.h	2016-07-18 08:28:04.000000000 +0300
+++ capstone/utils.h			2017-02-12 15:54:45.000000000 +0300
@@ -4,7 +4,11 @@
 #ifndef CS_UTILS_H
 #define CS_UTILS_H
 
+#if defined(CAPSTONE_HAS_OSXKERNEL)
+#include <libkern/libkern.h>
+#else
 #include <stddef.h>
+#endif
 #include "include/capstone.h"
 #include "cs_priv.h"
 
